### PR TITLE
Dev thumbs

### PR DESCRIPTION
**Pull request by briankassouf**

### CODE REVIEW

File: imagestore/s3store.go [hunk 1]
```diff
@@ -50,6 +51,16 @@ func (this *S3ImageStore) Save(src io.Reader, obj *StoreObject) (*StoreObject, e
 	return obj, nil
 }
 
+func (this *S3ImageStore) Get(obj *StoreObject) (io.Reader, error) {
+	bucket := this.client.Bucket(this.bucketName)
+	data, err := bucket.Get(this.toPath(obj))
```
Comment (jacobgreenleaf) [inline imagestore/s3store.go]: `bucket.GetReader`

**Comment by jacobgreenleaf** [comment]
Incidentally, since you're editing this code, you should change the S3ImageStore to use `PutReader` as well so we stream the data to S3 without allocating lots of memory.

File: server/server.go [hunk 2]
```diff
@@ -353,11 +410,31 @@ func (s *Server) Configure(muxer *http.ServeMux) {
...
 	router.HandleFunc("/", rootHandler)
 
 	muxer.Handle("/", router)
 }
 
+func (s *Server) buildThumbResponse(upload *uploadedfile.UploadedFile) (map[string]interface{}, error) {
+	factory := imagestore.NewFactory(s.Config)
+	thumbsResp := map[string]interface{}{}
+
+	for _, t := range upload.GetThumbs() {
```
Comment (jacobgreenleaf) [inline server/server.go]: potential future optimization: upload thumbs in parallel

**Comment by jacobgreenleaf** [comment]
:+1:

File: imagestore/s3store.go (hunk 1 above)
Comment (briankassouf) [inline imagestore/s3store.go]: good thought. Might need to refactor to return an io.ReadCloser though

File: server/server.go (hunk 2 above)
Comment (briankassouf) [inline server/server.go]: :+1:

**Pull request merged**